app/src/main/java/com/example/music_app/MainActivityViewModel.kt

```diff
@@ -25,29 +26,41 @@ class MainActivityViewModel(
     val isSplashScreenVisible: StateFlow<Boolean> = _isSplashScreenVisible.asStateFlow()
     private val _route: MutableStateFlow<Screens?> = MutableStateFlow(null)
     val route: StateFlow<Screens?> get() = _route
+    private val _error: MutableStateFlow<AppErrors?> = MutableStateFlow(null)
+    val error: StateFlow<AppErrors?> get() = _error
 
     init {
         checkStatus()
     }
 
-    private fun checkStatus() {
+    fun checkStatus() {
         viewModelScope.launch {
             isAuthorizedCheckUseCase().onSuccess { isAuth ->
-                when {
-                    !isAuth -> _route.value = Screens.LoginScreen
-                    isOutdatedCheckUseCase().component1()?: false -> {
-                        requestRefreshTokenUseCase().collect { token ->
-                            if (token.component1() != null) {
-                                _route.value =
-                                    Screens.PlaylistsScreen
+                if (isAuth) {
+                    isOutdatedCheckUseCase().onSuccess { isOutdated ->
+                        if (isOutdated) {
+                            requestRefreshTokenUseCase().collect { token ->
+                                token.onSuccess {
+                                    _route.value =
+                                        Screens.PlaylistsScreen
+                                }.onFailure { tokenRequestError ->
+                                    _error.value = tokenRequestError
+                                }
                             }
+                        } else {
+                            _route.value = Screens.PlaylistsScreen
                         }
+                    }.onFailure { isOutdatedError ->
+                        _error.value = isOutdatedError
                     }
-                    else -> _route.value = Screens.PlaylistsScreen
+                } else {
+                    _route.value = Screens.LoginScreen
```

Comment:
I think this line could be extracted to the separate method with only destination screen being passed (e.g. `navigateToDestination(destination: Screens)`